File: src/scrambler.php
<?php
  $numMoves = 25;
  $moves = [];
  $turnTypes = [ "U", "R", "L", "D", "B", "F" ];
  $revTypes = [ " ", "2", "'" ];
  $type = $turnTypes[rand(0,5)];
  echo "\n";
  while(count($moves) < $numMoves)
    if($moves == [] || end($moves) != ($type = $turnTypes[rand(0,5)])) {
      $moves[] = $type;
      echo $type . $revTypes[rand(0,2)] . (count($moves)%5==0?"\n":"\t");
    }
  echo "\n";
?>

File: bin/scrambler.rb
turnTypes = [ "U", "R", "L", "D", "B", "F" ]
revTypes = [ " ", "2", "'" ]
prng = Random.new
moves, numMoves, type = [], 25, turnTypes[prng.rand(6)]
puts
while moves.length < numMoves do
  if moves == [] or moves.last != (type = turnTypes[prng.rand(6)]) then
    moves << type
    print type + revTypes[prng.rand(3)] + (moves.length%5==0?"\n":"\t")
  end
end
puts
